src/tests/ftest/util/test_utils.py

```diff
@@ -0,0 +1,916 @@
+            self.written_data.append(TestContainerData())
+            self.written_data[-1].write_object(
+                self, self.record_qty.value, self.akey_size.value,
+                self.dkey_size.value, self.data_size.value, rank, obj_class)
+
+    @fail_on(DaosTestError)
+    def read_objects(self):
+        """Read the objects from the container and verify they match.
+
+        Returns:
+            bool: True if
+
+        """
+        self.open()
+        self.log.info("Reading objects in container %s", self.uuid)
+        status = len(self.written_data) > 0
+        for data in self.written_data:
+            status &= data.read_object(self)
+        return status
+
+    def execute_io(self, duration, rank=None, obj_class=None):
+        """Execute writes and reads for the specified time period.
+
+        Args:
+            duration (int): how long, in seconds, to write and read data
+            rank (int, optional): server rank. Defaults to None.
+            obj_class (int, optional): daos object class. Defaults to None.
+
+        Returns:
+            int: number of bytes written to the container
+
+        Raises:
+            DaosTestError: if there is an error writing, reading, or verify the
+                data
+
+        """
+        self.open()
+        self.log.info(
+            "Writing and reading objects in container %s for %s seconds",
+            self.uuid, duration)
+
+        total_bytes_written = 0
+        finish_time = time() + duration
+        while time() < finish_time:
+            self.written_data.append(TestContainerData())
+            self.written_data[-1].write_object(
+                self, 1, self.akey_size.value, self.dkey_size.value,
+                self.data_size.value, rank, obj_class)
+            total_bytes_written += self.data_size.value
+        return total_bytes_written
+
+    def get_target_rank_lists(self, message=""):
+        """Get a list of lists of target ranks from each written object.
+
+        Returns:
+            list: a list of list of targets for each written object in this
+                container
+
+        """
+        self.open()
+        target_rank_lists = []
+        for data in self.written_data:
+            try:
+                data.obj.get_layout()
+                # Convert the list of longs into a list of ints
+                target_rank_lists.append(
+                    [int(rank) for rank in data.obj.tgt_rank_list])
+            except DaosApiError as error:
+                raise DaosTestError(
+                    "Error obtaining target rank list for object {} in "
+                    "container {}: {}".format(data.obj, self.uuid, error))
+        if message is not None:
+            self.log.info("Target rank lists%s:", message)
+            for ranks in target_rank_lists:
+                self.log.info("  %s", ranks)
+        return target_rank_lists
+
+    def get_target_rank_count(self, target_rank_list, rank):
```

Comment:
(pylint-no-self-use) Method could be a function

Reply:
Fixed